visualizer/actionButton.js

```diff
@@ -20,15 +24,146 @@
     LEFTWARDS_ARROW: '\u2190'
   };
 
-  function createSemanticNameContainer() {
+  var deleteEntry = document.querySelector('#action.contextMenu #delete');
+  var editEntry = document.querySelector('#action.contextMenu #edit');
+
+  // Unselect all the semantics buttons, except the target semantic button
+  function unselectOtherSemanticButtons(targetNameContainer) {
+    var wrappers = document.querySelectorAll('#semantics .wrapper');
+    Array.prototype.forEach.call(wrappers, function(wrapper) {
+      var nameContainer = wrapper.querySelector('textarea.opName');
+      if (targetNameContainer === nameContainer) {
+        return;
+      }
+      wrapper.classList.remove('selected');
+
+      // Hide the argument list if there is one
+      var arrowButton = wrapper.querySelector('.arrow');
+      if (arrowButton) {
+        arrowButton.innerHTML = UnicodeChars.BLACK_DOWN_POINTING_TRIANGLE;
+        wrapper.querySelector('.arguments').hidden = true;
+      }
+    });
+  }
+
+  function retrieveArgs(wrapper) {
+    var argList = wrapper.querySelector('.arguments');
+    var nameValPairs = wrapper.querySelectorAll('.nameValuePair');
+    var args = Object.create(null);
+    Array.prototype.forEach.call(nameValPairs, function(nameValPair) {
+      var argNameContainer = nameValPair.querySelector('.name');
+      var argName = argNameContainer.value;
+
+      var valueContainer = nameValPair.querySelector('.value');
+      var value = valueContainer.value;
+      if (argName && /^[_a-zA-Z0-9]+$/.test(argName) && !(argName in args)) {
+        args[argName] = eval(value);  // eslint-disable-line no-eval
```

Comment:
So, funny things happen if you use `args` as the value of an argument :-)

As I mentioned in the last PR, it's usually slightly better to use `new Function` rather than `eval`, because `eval` gives access to all the variables in the scope where it is called.